spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageProtocolResolver.java

```diff
@@ -45,8 +45,9 @@
 
 	private ConfigurableListableBeanFactory beanFactory;
 
-	private GoogleStorageProtocolResolverSettings settings;
-	private boolean settingsAvailable = true;
+	private GoogleStorageProtocolResolverSettings settings
+			= new GoogleStorageProtocolResolverSettings();
+	private boolean checkedIfSettingsBeanAvailable;
```

Comment:
To be honest, my previous comment was exactly about the fact that we don't need extra variable.
We definitely need to load `storage` and `settings` beans only once. 
Since `storage` is mandatory, there is just enough to pull the `getBean(GoogleStorageProtocolResolverSettings.class)` just after `getBean(Storage.class)` in the same `if` block.

Does it make sense for you?

Also, please, consider to add blank lines around class properties.